Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/AddDocStrucTypeDialog.java

```diff
@@ -426,6 +424,10 @@ public void prepare() {
         }
         this.prepareDocStructTypes();
         prepareSelectPageOnAddNodeItems();
+        // reset type selection if previous selection is not allowed on current parent/position
+        if (getDocStructAddTypeSelectionItems().stream().noneMatch(a -> a.getValue().equals(docStructAddTypeSelectionSelectedItem))) {
```

Comment:
Is it possible to use a more meaningful variable name than `a` inside the expression? Maybe move this whole complex condition check outside the if condition itself like

```java 
bool dontRemberLastSelectedItem = getDocStructAddTypeSelectionItems()
   .stream()
   .noneMatch(item -> item.getValue().equals(docStructAddTypeSelectionSelectedItem));

if (dontRemberLastSelectedItem) {
   docStructAddTypeSelectionSelectedItem = "";
}
```

Reply:
I agree that `a` is not a very good variable name and that it should be renamed. 

Assigning the result of the stream expression to a new variable that is in turn immediately used just once in the next step does not really lower the complexity of the stream expression, though, and I wouldn't see the gain of introducing such a new variable. I would therefor suggest to keep the code as is and just rename the existing variable `a` within the stream to something more meaningful.

Reply:
I would agree with you if the condition itself is short and good readable. But this condition expression is long, complex and not good readable on the first look. I prefer to avoid this but if this is okay then it go.

Reply:
@henning-gerhardt I understand what you mean. You have a point because the expression is used in a condition check, which makes it a little less readable. If this was a case where the additional variable would just be returned in the next step, I would argue that it is indeed redundant, but in this case it comes down to personal preference, I guess. Would you mind re-reviewing the PR if you can live with this compromise?